### PR TITLE
add check for http:// URLs in easyconfig files added/changed in PRs

### PR DESCRIPTION
**Pull request by boegel**
works as expected, see #8999 where I squeezed in this change too by matter of proof of concept...

### CODE REVIEW

**Comment by boegel** [comment]
@Micket Once this is merged, tests in Travis will fail for (new/updated) PRs if easyconfigs still have `http://` in them.

We may need to whitelist for some stuff where `https://` doesn't work (should we auto-check if the specified URLs exist?)

**Comment by Micket** [comment]
@boegel Having looked through a bunch of them the past week, I've vaguely recall seeing at least a couple of exceptions already. I think autochecking would be a good idea before requiring it
If they aren't using https, and there is a positive response from `requests.get(url, verify=True)`, then they should prefer it.

**Comment by boegel** [comment]
@Micket Do you happen to recall an example of an exception, to test with?

**Comment by Micket** [comment]
https://github.com/easybuilders/easybuild-easyconfigs/pull/8997/files

**Comment by boegel** [comment]
enhanced test works as expected, see #8999 and #9006

File: test/easyconfigs/easyconfigs.py [hunk 1]
```diff
@@ -453,6 +453,28 @@ def check_sanity_check_paths(self, changed_ecs):
 
         self.assertFalse(failing_checks, '\n'.join(failing_checks))
 
+    def check_https(self, changed_ecs):
+        """Make sure https:// is used for homepage/source_urls (rather than http://)."""
+
+        whitelist = []
+
+        http_regex = re.compile('http://[^"\'\n]*', re.M)
```
Comment (wpoely86) [inline test/easyconfigs/easyconfigs.py]: + instead of * ?

File: test/easyconfigs/easyconfigs.py [hunk 2]
```diff
@@ -453,6 +453,28 @@ def check_sanity_check_paths(self, changed_ecs):
 
         self.assertFalse(failing_checks, '\n'.join(failing_checks))
 
+    def check_https(self, changed_ecs):
+        """Make sure https:// is used for homepage/source_urls (rather than http://)."""
```
Comment (wpoely86) [inline test/easyconfigs/easyconfigs.py]: add: if it exists?

File: test/easyconfigs/easyconfigs.py (hunk 1 above)
Comment (boegel) [reply test/easyconfigs/easyconfigs.py]: You mean `+`?

**Pull request merged**